File: matrix/diagonal_matrix.cc
// program for diagonal matrix

/*
15 0 0 0 0 
0 30 0 0 0 
0 0 45 0 0 
0 0 0 60 0 
0 0 0 0 75 
*/

#include <iostream>
using namespace std;

struct Matrix{
    int array[10];
    int dimension;
};

// set the value in matrix
void set(struct Matrix *m, int row, int column, int value){
    // formula is M[i,j] = non-zero if i == j
    if(row == column){
        m->array[row-1] = value;
    }
}

// get the value from matrix
int get(struct Matrix m, int row, int column){
    if(row == column){
        return m.array[row-1];
    }
    return 0;
}

// display the matrix
void display(struct Matrix m){
    for(int i = 0; i < m.dimension; i++){
        for(int j = 0; j < m.dimension; j++){
            if(i == j){
                cout << m.array[i] << " ";
            }
            else {
                cout << "0 ";
            }
        }
        cout << endl;
    }
}

int main(){
    struct Matrix m;
    m.dimension = 5;
    set(&m, 1,1,15);
    set(&m, 2,2,30);
    set(&m, 3,3,45);
    set(&m, 4,4,60);
    set(&m, 5,5,75);

    display(m);
    cout << get(m, 1,1) << endl;
}

File: matrix/diagonal_matrix.py
# Program for diagonal matrix

# set the value in matrix
def Set(matrix, row, column, value):
    # M[i,j] = non-zero if i == j
    if(row == column):
        matrix.append(value)
    
# get the value from matrix
def Get(matrix, row, column):
    if (row == column):
        return matrix[row-1]
    return None

# display the entire matrix
def display(matrix, dimension):
    for i in range(0, dimension):
        for j in range(0, dimension):
            if (i == j):
                print(matrix[i], end=' ')
            else:
                print('0 ', end=' ')
        print() # new line

if __name__ == '__main__':
    matrix = []
    dimension = 5
    Set(matrix, 1, 1, 15)
    Set(matrix, 2, 2, 30)
    Set(matrix, 3, 3, 45)
    Set(matrix, 4, 4, 60)
    Set(matrix, 5, 5, 75)

    display(matrix, dimension)
